schematic/models/metadata.py

```diff
@@ -235,3 +237,45 @@ def populateModelManifest(self, title, manifestPath: str, rootNode: str) -> str:
         emptyManifestURL = mg.get_manifest()
 
         return mg.populate_manifest_spreadsheet(manifestPath, emptyManifestURL)
+
+
+    def submit_metadata_manifest(self, manifest_path: str, dataset_id: str, validate_component: str, json_schema: str = None) -> bool:
+        """Wrap methods that are responsible for validation of manifests for a given component, and association of the 
+        same manifest file with a specified dataset.
+
+        Args:
+            manifest_path: Path to the manifest file, which contains the metadata.
+            dataset_id: Synapse ID of the dataset on Synapse containing the metadata manifest file.
+            validate_component: Component from the schema.org schema based on which the manifest template has been generated.
+
+        Returns:
+            True: If both validation and association were successful.
+            False: If the function could not successfully validate (and hence associate) the manifest file with the dataset.
+        """
+        # list of all the validation errors present in the manifest file
+        val_errors = []
+
+        # JSON schema need not be provided always, it is an optional argument
+        # if JSON schema is provided, then use it
+        if json_schema:
+            val_errors = self.validateModelManifest(manifestPath=manifest_path, rootNode=validate_component, jsonSchema=json_schema)
+
+        # automatic JSON schema generation and validation with that JSON schema
+        val_errors = self.validateModelManifest(manifestPath=manifest_path, rootNode=validate_component)
```

Comment:
Perhaps put in an else branch

Reply:
See [L#285](https://github.com/Sage-Bionetworks/schematic/blob/develop-submit_metadata_manifest/schematic/models/metadata.py#L285).